src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -367,6 +388,175 @@ ErrorIfModifyQueryNotSupported(Query *queryTree)
 
 
 /*
+ * Fails if the expression contains STABLE functions which accept any paramaters derived
+ * from a Var. Assumes the expression contains no VOLATILE functions.
+ *
+ * Var's are allowed, but only if they are passed solely to IMMUTABLE functions
+ */
+static bool ContainsDisallowedFunctionCalls(Node *expression)
+{
+	bool unused = false;
+	return ContainsDisallowedFunctionCallsWalker(expression, &unused);
+}
+
+
+static bool ContainsDisallowedFunctionCallsWalker(Node *expression, bool *containsVar)
+{
+	char volatileFlag = 0;
+	bool childContainsVar = false;
+	bool containsDisallowedFunction = false;
+
+	if (expression == NULL)
+	{
+		return false;
+	}
+
+	if (IsA(expression, Var))
+	{
+		*containsVar = true;
+		return false;
+	}
+
+	/* this list of nodes is taken from contain_mutable_functions_walker */
+	if (IsA(expression, OpExpr))
+	{
+		OpExpr *expr = (OpExpr *) expression;
+
+		set_opfuncid(expr);
+		volatileFlag = func_volatile(expr->opfuncid);
+	}
+	else if (IsA(expression, FuncExpr))
+	{
+		FuncExpr *expr = (FuncExpr *) expression;
+
+		volatileFlag = func_volatile(expr->funcid);
+	}
```

Comment:
[Sure it is](https://github.com/citusdata/citus/pull/488/files#diff-7393a721dde349871f266a5a2de89455R354), it's done by [this check](https://github.com/citusdata/citus/pull/488/files#diff-4c4f0976727d06684376ecb22e8f5362R524). If the current iteration is dealing with a STABLE function it recurses and fails if any descendants were a Var.